### DISCUSSION TITLE
How can i create an NPC (player or entity)

### DISCUSSION THREAD

**Discussion by DasLixou** [Minestom/Minestom#1014] [Help] [answered]
title

**Comment by ZakShearman** [comment, marked as answer]
You can simply follow the examples from the wiki: https://wiki.minestom.net/feature/entities#examples

They show everything you should need. It may also be useful to note that there are Entity#lookAt and Pos#withLookAt methods if you want them to look towards something specific.

**Reply by DasLixou** [reply]
It works, but VillagerMeta only sets the custom name, professtion etc. doesnt seem to work :c
```kotlin
it.customName = Component.text("Test")
it.isCustomNameVisible = true
it.villagerData.type = VillagerMeta.Type.DESERT
it.villagerData.level = VillagerMeta.Level.MASTER
it.villagerData.profession = VillagerMeta.Profession.FARMER
```

**Reply by DasLixou** [reply]
ok nvm solved
